### PR TITLE
Minor refactor of LangChain utilities/bridge to support LC ChatMessage

### PR DESCRIPTION
**Pull request by VKudlay**
# Description

some refactoring to support ChatMessage/ChatMessageChunks for interfaces like ChatNVIDIA, ChatOpenAI, etc.
@logan-markewich 

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] I stared at the code and made sure it makes sense
https://colab.research.google.com/drive/1LlAYuRfB1NlKJfusYOj7fJXB8nnxTMiP?usp=sharing

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by logan-markewich — APPROVED** [review]

**Comment by VKudlay** [comment]
I see the linting issue. Getting poetry set up...

**Comment by VKudlay** [comment]
@logan-markewich Hey Logan. Sorry, I canceled the auto-merge thing (didn't realize it was happening actually) since I wanted to fix the lint issue. Tests and lint should all pass now! (And this is all a short-term fix, of course. There are several deprecations which still work, but those can be ironed out later).

**Comment by logan-markewich** [comment]
@VKudlay auto-merge would have only ran once CI passed + someone approved, but no worries, sounds good 👍🏻 Will merge assuming CI passes

**Pull request merged**
